### DISCUSSION TITLE
Data traffic is not encapsulated with ESP

### DISCUSSION THREAD

**Discussion by BSaloni** [strongswan/strongswan#765] [Q&A]
Hi,

We are running a scenario using Strongswan as a UE side and server. We have created IPSec Signaling tunnel that we are using for Control Plane and that is perfectly working fine. We have created one more tunnel for User Plane and we want to use Child SA Tunnel(User Plane tunnel) for sending and receiving user traffic but the problem we are facing is data traffic going is GRE encapsulated but not ESP encapsulated.  Packet is reachable but not ESP encapsulated.
We want user traffic to be ESP encapsulated and packet should go  through CHILD SA tunnel.

Topology:
Control Plane:
10.0.0.9---192.168.56.102======192.168.56.107-10.0.0.10
User Plane:
60.60.0.3---192.168.56.102======192.168.56.107-10.0.0.11

XFRM policy:
sudo ip xfrm policy
src 60.60.0.3/32 dst 10.0.0.11/32 proto gre
        dir out priority 366975
        mark 0x43/0xffffffff
        tmpl src 192.168.56.102 dst 192.168.56.107
                proto esp spi 0x678fa0fc reqid 2 mode tunnel
src 10.0.0.11/32 dst 60.60.0.3/32 proto gre
        dir fwd priority 366975
        mark 0xc/0xffffffff
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 2 mode tunnel
src 10.0.0.11/32 dst 60.60.0.3/32 proto gre
        dir in priority 366975
        mark 0xc/0xffffffff
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 2 mode tunnel
src 10.0.0.9/32 dst 10.0.0.10/32 proto tcp
        dir out priority 366975
        mark 0x43/0xffffffff
        tmpl src 192.168.56.102 dst 192.168.56.107
                proto esp spi 0xc598c78a reqid 1 mode tunnel
src 10.0.0.10/32 dst 10.0.0.9/32 proto tcp
        dir fwd priority 366975
        mark 0xc/0xffffffff
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 1 mode tunnel
src 10.0.0.10/32 dst 10.0.0.9/32 proto tcp
        dir in priority 366975
        mark 0xc/0xffffffff
        tmpl src 192.168.56.107 dst 192.168.56.102

Please let me know if any more info requires

Regards
SaLoni

**Comment by tobiasbrunner** [comment]
You have marks set on these policies, so make sure that appropriate marks are applied to your traffic or don't configure any marks if they are not needed (and, of course, make sure that the traffic matches the policies in the first place).

**Reply by BSaloni** [reply]
Hi,

Thanks for the reply.

I have configured XFRM policy without any mark but still cannot see ping packet initiating from strongswan is not ESP encapsulated.

Please find output:
sudo ip xfrm state
[sudo] password for ubuntu:
src 192.168.56.102 dst 192.168.56.107
        proto esp spi 0x0e559501 reqid 2 mode tunnel
        replay-window 0 flag af-unspec
        auth-trunc hmac(sha1) 0x91e47af83cb6372ed74303e6b1ee757b48eb66b4 96
        enc cbc(aes) 0xfaa0dab8e1317fc9f66f325fa8ba86af4473762c0c99d3ba30578580804a23fe
        anti-replay context: seq 0x0, oseq 0x0, bitmap 0x00000000
src 192.168.56.107 dst 192.168.56.102
        proto esp spi 0xc8343aa0 reqid 2 mode tunnel
        replay-window 32 flag af-unspec
        auth-trunc hmac(sha1) 0x0ddd9b4d318df9f82c081b497d6216b0b9bf474d 96
        enc cbc(aes) 0x9b5bae219446ae3c6bd894da14d0abc0dd3a37774fea22247d06703b028e6df5
        anti-replay context: seq 0x0, oseq 0x0, bitmap 0x00000000
src 192.168.56.102 dst 192.168.56.107
        proto esp spi 0xcb9cb8be reqid 1 mode tunnel
        replay-window 0 flag af-unspec
        auth-trunc hmac(sha1) 0x8442602f5b6f382f306267ed95bb82e9ae2afb59 96
        enc cbc(aes) 0xf7fd9a8bbfa83463f780cb9628c1fef8
        anti-replay context: seq 0x0, oseq 0x121, bitmap 0x00000000
src 192.168.56.107 dst 192.168.56.102
        proto esp spi 0xcb9cb8be reqid 1 mode tunnel
        replay-window 32 flag af-unspec
        auth-trunc hmac(sha1) 0xbb52465c648a0957856fba440e35e91b31393054 96
        enc cbc(aes) 0xbfdc8acd64c0de33ce82a9082affcd3c
        anti-replay context: seq 0x120, oseq 0x0, bitmap 0xffffffff



 sudo ip xfrm policy
src 60.60.0.3/32 dst 10.0.0.11/32 proto gre
        dir out priority 366975
        tmpl src 192.168.56.102 dst 192.168.56.107
                proto esp spi 0x0e559501 reqid 2 mode tunnel
src 10.0.0.11/32 dst 60.60.0.3/32 proto gre
        dir fwd priority 366975
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 2 mode tunnel
src 10.0.0.11/32 dst 60.60.0.3/32 proto gre
        dir in priority 366975
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 2 mode tunnel
src 10.0.0.9/32 dst 10.0.0.10/32 proto tcp
        dir out priority 366975
        tmpl src 192.168.56.102 dst 192.168.56.107
                proto esp spi 0xcb9cb8be reqid 1 mode tunnel
src 10.0.0.10/32 dst 10.0.0.9/32 proto tcp
        dir fwd priority 366975
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 1 mode tunnel
src 10.0.0.10/32 dst 10.0.0.9/32 proto tcp
        dir in priority 366975
        tmpl src 192.168.56.107 dst 192.168.56.102
                proto esp reqid 1 mode tunnel


Regards
SaLoni

**Reply by tobiasbrunner** [reply]
Please read [RouteBasedVPN](https://wiki.strongswan.org/projects/strongswan/wiki/RouteBasedVPN#GRE) and if that doesn't help #196.
